docs/source/python/compute.rst

```diff
@@ -292,4 +292,71 @@ The resulting dataset will be an :class:`.InMemoryDataset` containing the joined
    animal: [["Brittle stars",null,null]]
    n_legs: [[5,null,null]]
 
+.. _py-filter-expr:
 
+Filtering by Expressions
+========================
+
+:class:`.Table` and :class:`.Dataset` can
+both be filtered using a boolean :class:`.Expression`.
+
+The expression can be built starting from a 
+:func:`pyarrow.field`. Comparisons and transformations
```

Comment:
That's a different field ... (maybe unfortunately, not sure if we should think about allowing that field object as well in the compute layer), but so pyarrow.field and pyarrow.compute.field are two different things
```suggestion
:func:`pyarrow.compute.field`. Comparisons and transformations
```

Reply:
Oh! Good catch. `pyarrow.field` is actually the factory for `pyarrow.Field`.